micropipenv.py

```diff
@@ -986,6 +1001,7 @@ def _iter_index_entry_str(
         return None
 
     for source in sections["sources"]:
+        source["url"] = resolve_nested_variables(source["url"])
```

Comment:
Do we want to have an option to control injecting environment variables using a micropipenv specific environment variable? I could imagine a use-case when micropipenv would be used for installing dependencies in a CI system and an attacker could adjust files to reveal sensitive environment variables to CI logs.

Reply:
Interesting point of view. Do you mean something like `MICROPIPENV_ENABLE_ENV_VARS`? And only if the mentioned variable is defined this function would do something with the variables in the files?

I must admit that it sounds reasonable but also kinda complicated for a user. Do you have an example of a tool doing something similar?

Reply:
> Interesting point of view. Do you mean something like `MICROPIPENV_ENABLE_ENV_VARS`? And only if the mentioned variable is defined this function would do something with the variables in the files?

Yes, in that case, the software could be more secure (inverted logic could also work fine but would require a proper setup by the CI system). 

> I must admit that it sounds reasonable but also kinda complicated for a user. Do you have an example of a tool doing something similar?

None I could think of from the top of my head. Just an idea during the review.

Reply:
I'm still thinking about how the additional option to enable/disable the feature can make it more secure. Let's say I'm an attacker and my plan is to modify Pipfile or Pipfile.lock to show me some secrets in CI logs. If I'm able to do that, I'm also able to modify the rest of the application files (because they are usually sent from the same place together) and use another way to uncover the secrets or adjust settings to enable the feature we are talking about.

I'll try to discuss it with my team today. They might have some experience with that.

Reply:
One more note from our morning brainstorming: If a user uses a CI/CD system for an open-source project, they should make sure that secrets in these services are available only for a run from protected branches because everybody can open a PR for your project and a random PR should not have access to any secrets because only some deploy phase needs them.

If this is true and the CI system is properly configured it would need an attacker's change to go through review to get it to a protected branch to have any impact. And in a case like that, an attacker can also modify other files, enable other insecure features and do a lot of stuff we cannot handle and it's the responsibility of a maintainer to take care of that.

Reply:
Yes, you are right. Let's keep this configuration on CI side (similarly as `/ok-to-test`). Sorry for bringing confusion. 👍🏻